### PR TITLE
get and post methods now working.

### PR DESCRIPTION
**Pull request by TahiraM**
Refactored code and created an interface for csvtojsonconverter
If you go to the Post Path method and enter a file path for a csv file it will find that file and will pull the data and upload to database

### CODE REVIEW

File: InternProject.CsvFileConverter.Library/Stores/DealDataRepository.cs [hunk 1]
```diff
@@ -1,24 +1,88 @@
...
             return dealDataList;
         }
+
+        public async Task<IEnumerable<DealData>> SaveManyAsync(DealData[] dealDataList)
+        {
+            using (var db = _dbContextFactory.Create())
+            {
+                db.Database.EnsureCreated();
+
+                foreach (var dealData in dealDataList)
```
Comment (Justin-FrenchIndustries) [inline InternProject.CsvFileConverter.Library/Stores/DealDataRepository.cs]: There is common code here between the SaveMany and SaveManyAsync methods? You can probably pull out common bits into a private method?

File: InternProject.CsvFileConverter.WebApi/Controllers/DealsController.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+        [HttpPost("{path}")]
+        public void UploadFile([FromBody]string path)
+        {
+            
+            _csvToJsonConverter.Convert(path);
+        }
+
+        //// PUT api/values/5
```
Comment (Justin-FrenchIndustries) [inline InternProject.CsvFileConverter.WebApi/Controllers/DealsController.cs]: Unless you are going to come back to these, delete the commented code. You can get it back through source control if necessary

File: InternProject.CsvFileConverter.WebApi/Startup.cs [hunk 3]
```diff
@@ -44,4 +63,22 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
             app.UseMvc();
         }
     }
+
+    //public static class ServiceCollectionController
```
Comment (Justin-FrenchIndustries) [inline InternProject.CsvFileConverter.WebApi/Startup.cs]: Remove commented code if not required

**Comment by TahiraM** [comment]
I have edited the code again

**Pull request merged**